crates/bitcoin-da/src/service.rs

```diff
@@ -357,17 +395,25 @@ impl BitcoinService {
     #[instrument(level = "trace", fields(prev_utxo), ret, err)]
     pub async fn send_transaction_with_fee_rate(
         &self,
-        prev_utxo: Option<UTXO>,
+        mut prev_utxo: Option<UTXO>,
         da_data: DaData,
         fee_sat_per_vbyte: u64,
-    ) -> Result<TxWithId, anyhow::Error> {
+    ) -> Result<(Vec<Txid>, TxWithId), anyhow::Error> {
         let network = self.network;
 
         let da_private_key = self.da_private_key.expect("No private key set");
 
         // get all available utxos
         let utxos = self.get_utxos().await?;
 
+        // If Utxo has already been spent, fallback to get_prev_utxo
+        // Would happen in case of cpfp spending reveal output
+        if let Some(prev) = &prev_utxo {
+            if !utxos.iter().any(|utxo| utxo == prev) {
+                prev_utxo = self.get_prev_utxo().await?;
+            }
+        }
```

Comment:
(referring to my last comment) and if we do it this way, now we don't need to search in the utxo list, the monitoring chain becomes the source of truth for `prev_utxo`